index.js

```diff
@@ -3,102 +3,104 @@
  * @param {String[]} paths The keys of state object
  *    that will be store in history
  */
-var createHistory = function (paths) {
-  paths = paths || []
-
-  var undo = Symbol('undo')
-  var redo = Symbol('redo')
-
-  var key = 'undoable'
-  if (paths.length > 0) {
-    undo = Symbol('undo_' + paths.join('_'))
-    redo = Symbol('redo_' + paths.join('_'))
-
-    key += '_' + paths.join('_')
-  }
-
-  return {
-    module: function (store) {
-      var ignoreNext = false
-      store.on('@init', function (state) {
-        ignoreNext = true
-        var init = {}
-        init[key] = {
-          past: [],
-          present: filterState(paths, state),
-          future: []
-        }
-        return init
-      })
-
-      store.on('@changed', function (state) {
-        if (ignoreNext) {
-          ignoreNext = false
-          return
-        }
-
-        ignoreNext = true
-        var undoable = state[key]
-
-        delete state[key]
-
-        state = filterState(paths, state)
-        var past = undoable.past
-        var present = undoable.present
-
-        var changed = {}
-        changed[key] = {
-          past: [].concat(past, [present]),
-          present: state,
-          future: []
-        }
-
-        return changed
-      })
-
-      store.on(undo, function (state) {
-        ignoreNext = true
-
-        var undoable = state[key]
-        if (undoable.past.length === 0) return
-        delete state[key]
-
-        state = filterState(paths, state)
-
-        var before = undoable.past.pop()
-
-        before[key] = {
-          present: Object.assign({}, before),
-          past: undoable.past,
-          future: [].concat(undoable.future, [state])
-        }
-
-        return before
-      })
-
-      store.on(redo, function (state) {
-        ignoreNext = true
-
-        var undoable = state[key]
-        if (undoable.future.length === 0) return
-
-        delete state[key]
-
-        state = filterState(paths, state)
-
-        var next = undoable.future.pop()
-
-        next[key] = {
-          present: Object.assign({}, next),
-          past: [].concat(undoable.past, [state]),
-          future: undoable.future
-        }
-
-        return next
-      })
-    },
-    UNDO: undo,
-    REDO: redo
+module.exports = {
+  createHistory: function (paths) {
+    paths = paths || []
```

Comment:
You can remove it and ask users always pass an array. It will be not a problem for most of the cases and will reduce the size.

Reply:
How I can ask the users? By definition in JSDoc? or in README?

Reply:
By `README.md`. ANyway they will copy-paste the example.

If you want better DX, you can do:

```js
if (process.env.NODE_ENV === 'development') {
  if (!paths) {
    throw new Error('better error text')
  }
}
```

Reply:
Ok, I've removed the `paths = paths || []` and add the error as you proposed. Please confirmed and I will merge this PR and release a new version.